x-pack/legacy/plugins/lens/server/usage/types.ts

```diff
@@ -4,11 +4,27 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export interface LensUsage {
-  visualization_types_overall: Record<string, number>;
-  visualization_types_last_30_days: Record<string, number>;
-  visualization_types_last_90_days: Record<string, number>;
-  saved_total: number;
-  saved_last_30_days: number;
-  saved_last_90_days: number;
+export interface LensTelemetryState {
+  runs: number;
+  byDate: Record<string, Record<string, number>>;
+  suggestionsByDate: Record<string, Record<string, number>>;
+  saved: LensVisualizationUsage;
 }
+
+export interface LensVisualizationUsage {
+  saved_overall: Record<string, number>;
```

Comment:
Just to be sure this can't explode in size, we should double-check that our generated event names have a realistically small potential set of values.

Reply:
These aren't generated names, they're all strings in code